aesara/tensor/basic_opt.py

```diff
@@ -1610,7 +1604,17 @@ def dimshuffled_alloc(i):
                         cond.append(eq(i_shp, cmp_shp))
                 if cond:
                     assert_op_in = assert_op(assert_op_in, *cond)
-            new_i.append(i.owner.inputs[0])
+            alloc_input = i.owner.inputs[0]
+            if alloc_input.ndim != i.ndim:
+                # The `Alloc` can add dimensions to the value.
+                # We replace those cases with a `DimShuffle` here.
```

Comment:
This matters in some new edge cases I found while fixing the original issues. Basically it led to the same type of bug, we were returning a new "graph" that was exactly like the original one without realizing it, because we were not addressing all possible cases for an `Alloc` inside the `Elemwise`

Reply:
That set of comments needs to mention how this avoids producing the same graph; otherwise, it won't help future readers very much.

Reply:
The test is about removing `Alloc`s so this shouldn't sound too surprising. It is only surprising in the light of the bug.